src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/GlobalJsonReader.cs

```diff
@@ -161,17 +161,27 @@ internal static bool TryGetPathOfFileAbove(string file, DirectoryInfo startingDi
 
             using (var reader = new JsonTextReader(new StringReader(json)))
             {
-                if (!reader.Read() || reader.TokenType != JsonToken.StartObject)
+                // Read to the first {
+                while (reader.Read() && reader.TokenType != JsonToken.StartObject)
                 {
-                    return null;
                 }
 
+                if (reader.TokenType != JsonToken.StartObject)
+                {
+                    // Return null if no { was found
+                    return versionsByName;
```

Comment:
NIT: How do you feel about returning null directly to improve readability?

Reply:
Also, could you please check if we have a UT that hits this if?

Reply:
Yeah that's probably better.

I do have a unit test to check for an "empty" JSON file here:

https://github.com/NuGet/NuGet.Client/blob/336320cc799d5a4e36da439148523e5ab9e4ff1e/test/NuGet.Core.Tests/Microsoft.Build.NuGetSdkResolver.Test/GlobalJsonReader_Tests.cs#L280-L284

But its only looking at a file with `{ }`.  I'll update it to try more stuff so that this `if` is exercised.